x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/SecurityContext.java

```diff
@@ -145,6 +145,7 @@ public void executeAsUser(User user, Consumer<StoredContext> consumer, Version v
      * The original context is provided to the consumer. When this method returns, the original context is restored.
      */
     public void executeAfterRewritingAuthentication(Consumer<StoredContext> consumer, Version version) {
+        // TODO: bwc here for API key metadata changes? i.e. deserialize the raw bytes into Map for older node.
```

Comment:
I think BWC code is required here so that: if target node is an old version, rewrite the `roleDescriptor` and `limitedRoleDescriptors` to be `Map<String, Object>`.